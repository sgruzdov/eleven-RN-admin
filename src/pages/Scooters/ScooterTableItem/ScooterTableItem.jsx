import React, { useState } from 'react'
import { TableCell, Button, TableRow, Collapse, IconButton } from '@material-ui/core'
import KeyboardArrowDownIcon from '@material-ui/icons/KeyboardArrowDown'
import KeyboardArrowUpIcon from '@material-ui/icons/KeyboardArrowUp'
import GoogleMapReact from 'google-map-react'
import { GOOGLE_API_KEY } from '../../../assets/constants'
import Marker from '../../../components/Marker/Marker'
import PropTypes from 'prop-types'

import styles from '../Scooters.module.scss'

const ScooterTableItem = ({ scooter }) => {
    const [open, setOpen] = useState(false)

    return (
        <>
            <TableRow>
                <TableCell className={styles.scooter_cell}>
                    <IconButton
                        className="button_collapse"
                        aria-label="expand row"
                        size="small"
                        onClick={() => setOpen(!open)}
                    >
                        {open ? <KeyboardArrowUpIcon /> : <KeyboardArrowDownIcon />}
                    </IconButton>
                </TableCell>
                <TableCell className={styles.scooter_cell}>{scooter.scooterId}</TableCell>
                <TableCell className={styles.scooter_cell} align="center">
                    {scooter.charge}
                </TableCell>
                <TableCell className={styles.scooter_cell} align="center">
                    {scooter.active ? 'Активно' : 'Не активно'}
                </TableCell>
                <TableCell className={styles.scooter_cell} align="center">
                    {scooter.userActive ? 'Активно' : 'Не активно'}
                </TableCell>
                <TableCell className={styles.scooter_cell} align="right">
                    {scooter.breakdown ? 'Активно' : 'Не активно'}
                </TableCell>
            </TableRow>
            <TableRow>
                <TableCell style={{ padding: 0 }} colSpan={6}>
                    <Collapse in={open} timeout="auto" unmountOnExit className="collapse">
                        <div className={styles.scooter_cell_wrap}>
                            <div className={styles.scooter_cell_descr}>
                                <div className={styles.scooter_cell_descr_title}>Действия</div>

                                <div className={styles.scooter_cell_descr_buttons}>
                                    <Button color="primary" variant="outlined">
                                        Изменить статусы
                                    </Button>
                                    <Button color="secondary" variant="outlined">
                                        Удалить самокат
                                    </Button>
                                </div>
                            </div>
                            <div className={styles.scooter_cell_map}>
                                <GoogleMapReact
                                    bootstrapURLKeys={{ key: GOOGLE_API_KEY }}
                                    defaultCenter={[scooter.location.latitude, scooter.location.longitude]}
                                    defaultZoom={14}
                                >
                                    <Marker
                                        lat={scooter.location.latitude}
                                        lng={scooter.location.longitude}
                                        marker={scooter}
                                    />
                                </GoogleMapReact>
                            </div>
                        </div>
                    </Collapse>
                </TableCell>
            </TableRow>
        </>
    )
}

ScooterTableItem.propTypes = {
    scooter: PropTypes.shape({
        location: PropTypes.objectOf(PropTypes.number),
        _id: PropTypes.string,
        scooterId: PropTypes.number,
        charge: PropTypes.number,
        active: PropTypes.bool,
        userActive: PropTypes.bool,
        breakdown: PropTypes.bool,
        __v: PropTypes.number,
    }),
}

export default ScooterTableItem
