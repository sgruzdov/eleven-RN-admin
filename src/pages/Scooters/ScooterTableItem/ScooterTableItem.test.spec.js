import ScooterTableItem from './ScooterTableItem'

const scooter = {
    location: {
        latitude: 53.853253464874754,
        longitude: 27.601949922194905,
    },
    _id: '60f5b4bad21fa65a98c3f9f5',
    scooterId: 955227,
    charge: 62,
    active: true,
    userActive: true,
    breakdown: true,
    __v: 0,
}

describe('ScooterTableItem component', () => {
    let component

    beforeEach(() => {
        component = setUp(ScooterTableItem, { scooter })
    })

    it('should render ScooterTableItem component', () => {
        expect(component).toMatchSnapshot()
    })

    it('handle click collapse button', () => {
        const collapse = component.find('.collapse').prop('in')
        component.find('.button_collapse').simulate('click')
        expect(component.find('.collapse').prop('in')).toEqual(!collapse)
    })

    describe('correctly status scooter', () => {
        it('active', () => {
            scooter.active = false
            component = setUp(ScooterTableItem, { scooter })
            expect(toJson(component.find('.scooter_cell'))[3].children[0]).toEqual('Не активно')
        })

        it('userActive', () => {
            scooter.userActive = false
            component = setUp(ScooterTableItem, { scooter })
            expect(toJson(component.find('.scooter_cell'))[4].children[0]).toEqual('Не активно')
        })

        it('userActive', () => {
            scooter.breakdown = false
            component = setUp(ScooterTableItem, { scooter })
            expect(toJson(component.find('.scooter_cell'))[5].children[0]).toEqual('Не активно')
        })
    })
})
