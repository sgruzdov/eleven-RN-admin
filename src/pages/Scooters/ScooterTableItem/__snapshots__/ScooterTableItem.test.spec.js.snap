// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScooterTableItem component should render ScooterTableItem component 1`] = `
<Fragment>
  <WithStyles(ForwardRef(TableRow))>
    <WithStyles(ForwardRef(TableCell))
      className="scooter_cell"
    >
      <WithStyles(ForwardRef(IconButton))
        aria-label="expand row"
        className="button_collapse"
        onClick={[Function]}
        size="small"
      >
        <Memo(ForwardRef(KeyboardArrowDownIcon)) />
      </WithStyles(ForwardRef(IconButton))>
    </WithStyles(ForwardRef(TableCell))>
    <WithStyles(ForwardRef(TableCell))
      className="scooter_cell"
    >
      955227
    </WithStyles(ForwardRef(TableCell))>
    <WithStyles(ForwardRef(TableCell))
      align="center"
      className="scooter_cell"
    >
      62
    </WithStyles(ForwardRef(TableCell))>
    <WithStyles(ForwardRef(TableCell))
      align="center"
      className="scooter_cell"
    >
      Активно
    </WithStyles(ForwardRef(TableCell))>
    <WithStyles(ForwardRef(TableCell))
      align="center"
      className="scooter_cell"
    >
      Активно
    </WithStyles(ForwardRef(TableCell))>
    <WithStyles(ForwardRef(TableCell))
      align="right"
      className="scooter_cell"
    >
      Активно
    </WithStyles(ForwardRef(TableCell))>
  </WithStyles(ForwardRef(TableRow))>
  <WithStyles(ForwardRef(TableRow))>
    <WithStyles(ForwardRef(TableCell))
      colSpan={6}
      style={
        Object {
          "padding": 0,
        }
      }
    >
      <WithStyles(ForwardRef(Collapse))
        className="collapse"
        in={false}
        timeout="auto"
        unmountOnExit={true}
      >
        <div
          className="scooter_cell_wrap"
        >
          <div
            className="scooter_cell_descr"
          >
            <div
              className="scooter_cell_descr_title"
            >
              Действия
            </div>
            <div
              className="scooter_cell_descr_buttons"
            >
              <WithStyles(ForwardRef(Button))
                color="primary"
                variant="outlined"
              >
                Изменить статусы
              </WithStyles(ForwardRef(Button))>
              <WithStyles(ForwardRef(Button))
                color="secondary"
                variant="outlined"
              >
                Удалить самокат
              </WithStyles(ForwardRef(Button))>
            </div>
          </div>
          <div
            className="scooter_cell_map"
          >
            <t
              bootstrapURLKeys={
                Object {
                  "key": "AIzaSyDJdAJmdPxqJtHAsgMEKBFNZFFx5D6kIoA",
                }
              }
              debounced={true}
              defaultCenter={
                Array [
                  53.853253464874754,
                  27.601949922194905,
                ]
              }
              defaultZoom={14}
              distanceToMouse={[Function]}
              googleMapLoader={[Function]}
              heatmap={Object {}}
              heatmapLibrary={false}
              hoverDistance={30}
              layerTypes={Array []}
              options={[Function]}
              shouldUnregisterMapOnUnmount={true}
              style={
                Object {
                  "height": "100%",
                  "margin": 0,
                  "padding": 0,
                  "position": "relative",
                  "width": "100%",
                }
              }
              yesIWantToUseGoogleMapApiInternals={false}
            >
              <Marker
                lat={53.853253464874754}
                lng={27.601949922194905}
                marker={
                  Object {
                    "__v": 0,
                    "_id": "60f5b4bad21fa65a98c3f9f5",
                    "active": true,
                    "breakdown": true,
                    "charge": 62,
                    "location": Object {
                      "latitude": 53.853253464874754,
                      "longitude": 27.601949922194905,
                    },
                    "scooterId": 955227,
                    "userActive": true,
                  }
                }
              />
            </t>
          </div>
        </div>
      </WithStyles(ForwardRef(Collapse))>
    </WithStyles(ForwardRef(TableCell))>
  </WithStyles(ForwardRef(TableRow))>
</Fragment>
`;
